### PR TITLE
Main Pages and Navigation

### PR DESCRIPTION
**Pull request by robinson-md**
**This PR includes:** 

- Main Pages and Navigation system
- Styling and layout structure for Tabs, Dashboard, Profile, Todos & Notes
- ChartJS library integration
- Store Management /w NGXS (states & actions for profile, notes & todos)
- Simple Mocked back-end data section

### CODE REVIEW

File: src/pages/dashboard/dashboard.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  public chartClicked(e: any): void {
+    console.log(e);
+  }
+
+  public chartHovered(e: any): void {
+    console.log(e);
+  }
+
+  goSettings(): void {
+    this.navCtrl.setRoot(Pages.PROFILE_PAGE);
```
Comment (AhsanAyaz) [inline src/pages/dashboard/dashboard.ts]: We're going to profile page but naming the function `goSettings`. Could probably rename the function

File: src/pages/dashboard/dashboard.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+@IonicPage()
+@Component({
+  selector: 'page-dashboard',
+  templateUrl: 'dashboard.html',
+})
+export class DashboardPage implements MainPage {
+
+  // Doughnut
+  public doughnutChartLabels: string[] = ['Month', 'Week', 'Today'];
```
Comment (AhsanAyaz) [inline src/pages/dashboard/dashboard.ts]: `doughnutChartLabels` name could be renamed to something that depicts what that chart is meant to show.

File: src/pages/profile/profile.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      <button type="button" ion-button clear [ngClass]="{ 'active': selectedTab === profileFilters.note }" (click)="toogle(profileFilters.note)">
+        Notes
+      </button>
+    </div>
+  </div>
+
+  <!-- List -->
+  <ion-scroll scrollY="true" [ngStyle]="{ 'height.px': scrollHeight }">
+    <ion-list padding no-lines>
+      <list-item *ngFor="let item of (items$ | async)" [item]="item"></list-item>
```
Comment (AhsanAyaz) [inline src/pages/profile/profile.html]: Always a good idea to use `trackBy` function with the ngFor stuff for better performance

File: src/pages/todos/todos.html [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    </ion-buttons>
+    <ion-title>Todos</ion-title>
+  </ion-navbar>
+
+</ion-header>
+
+
+<ion-content padding-horizontal>
+  <ion-list>
+    <list-item *ngFor="let todo of (todos$ | async)?.todos" [item]="todo"></list-item>
```
Comment (AhsanAyaz) [inline src/pages/todos/todos.html]: using `trackBy` function with `*ngFor`

**Review by AhsanAyaz — CHANGES_REQUESTED** [review]

**Comment by robinson-md** [comment]
Hey @AhsanAyaz  I've applied the suggested changes.
Totally forgot about the trackBy, thanks man.

**Pull request merged**